src/monster.cpp

```diff
@@ -265,9 +265,12 @@ void monster::setpos( const tripoint &p )
     bool wandering = wander();
     g->update_zombie_pos( *this, p );
     position = p;
-    if( has_effect( effect_ridden ) && position != g->u.pos() ) {
-        add_msg( m_debug, "Ridden monster %s moved independently and dumped player", get_name() );
-        g->u.forced_dismount();
+    if( has_effect( effect_ridden ) && mounted_player ) {
```

Comment:
```suggestion
    if( has_effect( effect_ridden ) && mounted_player && mounted_player->pos() != pos() ) {
        add_msg(...);
        mounted_player ->forced_dismount();
```
No need to split the condition checking across multiple statements.